src/azul/drs.py

```diff
@@ -269,21 +266,30 @@ def get_object(self,
         depends on the access method specified.
         """
         if access_id is None:
```

Comment:
Consider calling `_get_url` here and pass the parsed URL instead. This would save 1 line of code :)

Reply:
To be totally clear, I mean calling `_get_url` before the `if`-`else`

Reply:
Unfortunately both `_get_object()` and `_get_object_access()` need to call `_get_url()` individually since `_get_object()` can obtain an `access_id` and then call `_get_object_access()` with it which requires a new URL being generated.

Reply:
Good catch.